packages/service-library/tests/test_archiving_utils.py

```diff
@@ -501,3 +519,28 @@ async def test_archive_unarchive_check_exclude(
     assert (
         relative_unarchived_paths == params.expected_result
     ), f"Exclude rules: {exclude_patterns=}"
+
+
+async def test_unarchive_dir_raises_error(
+    zipfile_single_file_extract_worker_raises_error: None,
+    dir_with_random_content: Path,
+    tmp_path: Path,
+):
+    temp_dir_one = tmp_path / "one"
+    temp_dir_two = tmp_path / "two"
+
+    temp_dir_one.mkdir()
+    temp_dir_two.mkdir()
+
+    archive_file = temp_dir_one / "archive.zip"
+
+    await archive_dir(
+        dir_to_compress=dir_with_random_content,
+        destination=archive_file,
+        store_relative_path=True,
+        compress=True,
+    )
+
+    with pytest.raises(ArchiveError) as err_info:
```

Comment:
suggestion: there is a match argument in ```pytest.raises``` for this check in the error

Reply:
@sanderegg added in #3348